### PR TITLE
Reduce state machine in AuthorizationMiddlewareResultHandler

### PR DESCRIPTION
**Pull request by Kahbazi**

### CODE REVIEW

File: src/Security/Authorization/Policy/src/AuthorizationMiddlewareResultHandler.cs [hunk 1]
```diff
@@ -12,41 +12,47 @@ namespace Microsoft.AspNetCore.Authorization.Policy;
...
-            }
-            else
-            {
-                await context.ChallengeAsync();
-            }
-
-            return;
+            return next(context);
+        }
+        else
```
Comment (Tratcher) [inline src/Security/Authorization/Policy/src/AuthorizationMiddlewareResultHandler.cs:21]: This else can be flattened?

**Review by Tratcher — APPROVED** [review]

File: src/Security/Authorization/Policy/src/AuthorizationMiddlewareResultHandler.cs (hunk 1 above)
Comment (Kahbazi) [reply src/Security/Authorization/Policy/src/AuthorizationMiddlewareResultHandler.cs]: Done!

**Comment by Kahbazi** [comment]
@Tratcher Was there a command to re-run the build pipeline? It seems the failure is not related to this PR.

**Comment by wtgodbe** [comment]
I just hit retry in azdo, which will re-run only the failing jobs. In general you can use "/a z p run"  (without the spaces in azp) to re-run the whole CI

**Pull request merged**